applications/CompressiblePotentialFlowApplication/custom_operations/potential_to_compressible_navier_stokes_operation.cpp

```diff
@@ -0,0 +1,117 @@
+}
+
+const Parameters PotentialToCompressibleNavierStokesOperation::GetDefaultParameters() const
+{
+    const Parameters default_parameters = Parameters(R"({
+        "origin_model_part"       : "",
+        "destination_model_part"  : "",
+        "reference_temperature"   : 273,
+        "compute_nodal_velocities": true
+    })");
+    return default_parameters;}
+
+void PotentialToCompressibleNavierStokesOperation::Execute()
+{
+    KRATOS_TRY;
+
+    const std::string origin_model_part_name = mParameters["origin_model_part"].GetString();
+    const std::string destination_model_part_name = mParameters["destination_model_part"].GetString();
+    const double& reference_temperature = mParameters["reference_temperature"].GetDouble();
+    const double& compute_nodal_velocities = mParameters["compute_nodal_velocities"].GetBool();
+    
+    // Saving the modelparts
+    auto& mOriginModelPart = mpModel->GetModelPart(origin_model_part_name);
+    auto& mDestinationModelPart = mpModel->GetModelPart(destination_model_part_name);
+
+    const int n_orig_nodes = mOriginModelPart.NumberOfNodes();
+    const int n_dest_nodes = mDestinationModelPart.NumberOfNodes();
+
+    // Check number of nodes
+    KRATOS_ERROR_IF_NOT(n_orig_nodes == n_dest_nodes)
+        << "Origin and destination model parts have different number of nodes."
+        << "\n\t- Number of origin nodes: " << n_orig_nodes
+        << "\n\t- Number of destination nodes: " << n_dest_nodes << std::endl;
+
+    // Getting the required variables
+    const double& gamma = mOriginModelPart.GetProcessInfo().GetValue(HEAT_CAPACITY_RATIO);
+    const double& sound_velocity = mOriginModelPart.GetProcessInfo().GetValue(SOUND_VELOCITY);
+    const double& free_stream_rho = mOriginModelPart.GetProcessInfo().GetValue(FREE_STREAM_DENSITY);
+    const double& free_stream_mach = mOriginModelPart.GetProcessInfo().GetValue(FREE_STREAM_MACH);
+    const double& specific_heat = (std::pow(sound_velocity,2) / (gamma * reference_temperature)) / (gamma - 1.0);
+    
+    // Calculate de nodal velocities
+    if (compute_nodal_velocities){
+        // Construct the ComputeNodalPotentialFlowVelocityProcess
+        const std::vector<std::string> variable_array = {"VELOCITY"};
+        ComputeNodalValueProcess ComputeNodalValueProcess(mOriginModelPart, variable_array);
+
+        // Execute the ComputeNodalPotentialFlowVelocityProcess
+        ComputeNodalValueProcess.Execute();}
+
+    // Transfer the potential flow values as initial condition for the compressible problem
+    IndexPartition<std::size_t>(n_orig_nodes).for_each([&](std::size_t index)
+                                                       {
+        auto it_dest_node = mDestinationModelPart.NodesBegin() + index;
+        const auto it_orig_node = mOriginModelPart.NodesBegin() + index;
+
+        // Getting potential flow velocities
+        const auto &velocity = it_orig_node->GetValue(VELOCITY);
```

Comment:
```suggestion
        const auto &r_velocity = it_orig_node->GetValue(VELOCITY);
```
See our style guide.